### PR TITLE
Feature/distance alpine ridge

### PR DESCRIPTION
**Pull request by louisPoulain**
Add new feature for Alpine ridge distance. The distance is now positive if the station is to the north of the ridge and negative otherwise.
The feature is tested via a new test, which allow visual evaluation of the sign of the distance.

### CODE REVIEW

**Comment by louisPoulain** [comment]
@dnerini Do you want to check the code before I merge ?

**Comment by dnerini** [comment]
thanks @louisPoulain ! I'd like to see a plot like the one below, where we see the values on a grid, would that be possible?

image

(this is the current version using the absolute distance only)

File: pyproject.toml [hunk 1]
```diff
@@ -12,6 +12,8 @@ pandas = "*"
 xarray = "*"
 pyproj = "*"
 bottleneck = "*"
+matplotlib = "*"
```
Comment (dnerini) [inline pyproject.toml:15]: these should be dev dependencies

File: mlpp_features/experimental.py [hunk 2]
```diff
@@ -47,18 +47,52 @@ def distance_point_to_segment(
...
 
 
+def signs_points_to_line(
+    points: List[Tuple[float, float]], line_points: List[Tuple[float, float]]
+) -> List[int]:
+    """For a list of points, find the sign of the distance to a polyline."""
+    signs = []
+    min_longitude = min([point[1] for point in line_points])
+    max_longitude = max([point[1] for point in line_points])
+    print(min_longitude, max_longitude)
```
Comment (dnerini) [inline mlpp_features/experimental.py:57]: either remove the print statement or use the logging module (but use a clearer log message in this case)

File: mlpp_features/experimental.py [hunk 3]
```diff
@@ -47,18 +47,52 @@ def distance_point_to_segment(
     return np.linalg.norm(point - closest_point)
 
 
+def signs_points_to_line(
+    points: List[Tuple[float, float]], line_points: List[Tuple[float, float]]
+) -> List[int]:
+    """For a list of points, find the sign of the distance to a polyline."""
```
Comment (dnerini) [inline mlpp_features/experimental.py:53]: can you provide more details about inputs and outputs? e.g., points is a tuple of (latitude, longitude)

File: mlpp_features/experimental.py (hunk 2 above)
Comment (dnerini) [reply mlpp_features/experimental.py:57]: this is true also for the other print statements

File: mlpp_features/experimental.py [hunk 4]
```diff
@@ -47,18 +47,52 @@ def distance_point_to_segment(
...
+    signs = []
+    min_longitude = min([point[1] for point in line_points])
+    max_longitude = max([point[1] for point in line_points])
+    print(min_longitude, max_longitude)
+    for point in points:
```
Comment (dnerini) [inline mlpp_features/experimental.py:58]: you could improve the readibility of this loop my modularizing the code a bit, e.g.

```
    for point in points:
        if point[1] < min_longitude:
            sign = get_sign_for_outside_point(point, min_longitude)
        elif point[1] > max_longitude:
            sign = get_sign_for_outside_point(point, max_longitude)
        else:
            sign = get_sign_for_inside_point(point)
        signs.append(sign)
```

**Review by dnerini — COMMENTED** [review]
Very nice work! I left few initial comments

File: mlpp_features/experimental.py (hunk 2 above)
Comment (louisPoulain) [reply mlpp_features/experimental.py:57]: Yes, I forgot to remove the prints which were there for debug purposes :)

**Comment by dnerini** [comment]
image

**Comment by louisPoulain** [comment]
The new changes are ready to be reviewed. Currently there is still the old function to calculate the distance and project to epsg:2056 but I don't need them anymore

File: mlpp_features/notebooks/visualization_alpine_ridge.ipynb
Comment (dnerini) [inline mlpp_features/notebooks/visualization_alpine_ridge.ipynb:1]: move folder to project root

Comment (louisPoulain) [reply mlpp_features/notebooks/visualization_alpine_ridge.ipynb:1]: Done

**Review by dnerini — APPROVED** [review]

**Pull request merged**